### ISSUE TITLE
Feature : Request to Add Restore Option for Deleted Files in Right-Click Menu

### ISSUE THREAD

**Issue by Lukiluc29** [files-community/Files#12403]
### What feature or improvement do you think would benefit Files?

I've noticed that when I accidentally delete a file, there is no option to restore it by right-clicking on it, similar to the Windows file explorer. Therefore, I believe it would be beneficial to integrate this option, if possible. It would be great to have options like "Restore Selected Item" or "Restore Moved Item" in the right-click menu.

### Requirements

Add the restore option to the right-click menu.

### Files Version

Version : 2.4.82.0

### Windows Version

Windows 11 22H2 22621.1702

**Comment by Josh65-2201** [comment]
Thanks for the feedback. Are you right clicking in the recycle bin as that's where the options are.

**Comment by Lukiluc29** [comment]
> Thanks for the report. I am unable to reproduce this, Can you give the following information.
> 
> * Are you right clicking in the recycle bin as that's where the options are.
> * What is the file/folder you are right clicking on?

I think I expressed myself badly. I would like you to add in the future an option that when you right click on an empty space after having moved or deleted an element that in the contextual menu there is the option "Restore the deleted element" or " Restore the moved element" as on the Windows file explorer.

**Comment by Josh65-2201** [comment]
Thanks for clarifying

**Comment by yaira2** [comment]
Does the ctrl + z shortcut work?

**Comment by yaira2** [comment]
@Lukiluc29

**Comment by Lukiluc29** [comment]
@yaira2 Yes, ctrl + z works fine but I prefer this function like in windows because I'm more used to it than the ctrl + z shortcut
